### PR TITLE
Add without-markers CLI option

### PR DESCRIPTION
**Pull request by wieczorek1990**
Some people prefer simple requirements.txt files without hashes and python version markers.
Added excluding markers.

### CODE REVIEW

**Comment by dimbleby** [comment]
A requirements.txt exported without markers is in general broken: installing from it may install the wrong things, may install incompatible things, may try to install two different versions of the same thing.

Who are these people who prefer "simple" requirements.txt?  What is the use case?

**Comment by wieczorek1990** [comment]
I do create those requirements.txt by hand.

W dniu sob., 2.07.2022 o 22:51 David Hotham ***@***.***>
napisał(a):

**Comment by dimbleby** [comment]
And what's the use case?  What problem do markers cause you?

To repeat myself, requirements.txt files produced by this MR can and will be broken.

If this is just a cosmetic preference: strongly recommend that you should prefer correct output over short output.

**Comment by wieczorek1990** [comment]
This output is a generated mess. I do write human readable files not generated mess.

**Comment by dimbleby** [comment]
I don't get to make the decisions; but I'd strongly suggest that this is not a good enough reason to produce broken output - and therefore this MR should be rejected.

You may find that when this repository catches up with the most recent poetry and (especially) poetry-core that there are improvements; the merging of markers has seen improvements.

Meanwhile if you want broken marker-free requirements.txt for your own purposes - suggest that you post-process the output produced here with a suitable `sed` command or similar.

**Comment by wieczorek1990** [comment]
I already post-process using a script, and wanted the flag for my process.
If you don't see somebody can have other process, then maybe don't publish
to common place.

**Comment by dimbleby** [comment]
It's precisely because this project has more than one user that it would be unwise to offer a flag that will, in a large number of cases, result in broken output.

The inevitable consequence of accepting this MR would be bug reports complaining that the output requirements.txt was uninstallable.  And they'd be right.

**Comment by wieczorek1990** [comment]
No, you don’t get customisation for a specific need.
There’s already —without-hashes and other options and this feature is
similar.
There’s many people who do mininal files and those with markers are just
unreadable for humans.
Pepole who upgrade regulary don’t use markers at all.
In your thinking we should also remove other format customisation flags.
This project as is is not an opinionated one-way, it’s customisable by
flags.

W dniu niedz., 3.07.2022 o 12:06 David Hotham ***@***.***>
napisał(a):

**Comment by dimbleby** [comment]
The enormous difference between `--without-hashes` and `--without-markers` is that files produced with `--without-hashes` are not broken, whereas files produced by `--without-markers` are broken.  

If your MR produced correct `requirements.txt` files I would have no problem with it.

**Comment by wieczorek1990** [comment]
No, —without-hahses produces not full features requiremets.txt. Therefore
some may say it’s broken.

W dniu niedz., 3.07.2022 o 12:20 David Hotham ***@***.***>
napisał(a):

**Comment by dimbleby** [comment]
Perhaps you don't understand the sense in which a marker-less requirements.txt is broken.

poetry's lockfile describes a solution that is suitable for all platforms, and the exported requirements.txt does the same.  Therefore it's not unusual to end up with an export that looks something like (simplified):
```
foo==1.0.0 ; python_version < 3.7
foo==2.0.0 ; python_version >= 3.7
bar==1.0.0 ; platform_system == "windows"
```
removing the markers gives:
```
foo==1.0.0
foo==2.0.0
bar==1.0.0
```
which will try to install two versions of `foo` everywhere, and try to install `bar` even on platforms where it may be uninstallable. 

This isn't just incomplete, this is badly broken.

An approach that might be more acceptable here - though I'm afraid, harder for you to do - would be to specify that you are exporting for "the current environment".  The resulting `requirements.txt` would unconditionally include everything that would be installed in the current python interpreter and platform etc.

That would at least produce a `requirements.txt` that was certain to work in the environment that exported it.

**Comment by wieczorek1990** [comment]
But I don’t want to use the markers feature.
I used to write requirements.txt by hand with only package and version exact match.
Now I want to use this plugin with the same simplicity in mind.
Try to install something older with markers syntax is not my use case. I want exact version.
Would use many files if using many environments instead of the markers feature.

**Comment by dimbleby** [comment]
Hopefully it is now clear to you that the markers in the files produced by this plugin are not an optional feature, they are essential to the correctness of the output.  Simply discarding them does not work.

I suppose you could provide command line parameters completely describing the target environment, and generate a `requirements.txt` meant for that target.

I can't believe that this would be worth the trouble, though.  You find the markers ugly, but by far the easiest thing for you to do would be to stop worrying about that!

**Comment by wieczorek1990** [comment]
So you guys did not go simplicity and configurability.

**Comment by wieczorek1990** [comment]
Markes add additional package information into the file which not everybody
is used to. I found it out only by using this package.

W dniu niedz., 3.07.2022 o 12:23 Łukasz Wieczorek ***@***.***>
napisał(a):

**Comment by neersighted** [comment]
I evidently missed the entire dialogue on this PR -- @wieczorek1990, not sure why you suddenly replied after closing it months ago?

Regardless, I am not in favor of this change -- @dimbleby is right that it will produce broken outputs. `poetry export` is meant to export a Poetry lockfile, not generate a flat list of requirements. If you're looking for the simple option, `pip freeze` is a better choice and will work on your (installed) project.

`poetry export` can't do the same without having to install your project (or learn how to parse tags itself), so it's output would be inferior to `pip freeze` even if we stripped markers, without a much more significant redesign and change in scope of the plugin.

To be honest the functionality requested here would make more sense in the `poetry bundle` plugin, only because it's already architected to 'render' the solution for the current platform already, where-as export is agnostic and tries to preserve as much information from the Poetry solution as possible.

In short, I don't think that there's a reason to implement this unless a large number of users show up asking for it -- I'd suggest opening a feature request issue so we can see if there is demand for such a thing.

**Comment by wieczorek1990** [comment]
Thanks for reply.

W dniu wt., 11.10.2022 o 12:48 Bjorn Neergaard ***@***.***>
napisał(a):
